src/app-layout/visual-refresh/trigger-button.tsx

```diff
@@ -32,38 +69,134 @@ function TriggerButton(
     ariaExpanded,
     ariaControls,
     onClick,
+    hasTooltip = false,
+    tooltipText,
     testId,
+    disabled = false,
+    isMobile = false,
     badge,
     selected = false,
     highContrastHeader,
   }: TriggerButtonProps,
   ref: React.Ref<ButtonProps.Ref>
 ) {
+  const containerRef = React.useRef(null);
+  const tooltipValue = tooltipText ? tooltipText : ariaLabel ? ariaLabel : '';
+  const [showTooltip, setShowTooltip] = useState<boolean>(false);
+
+  const onShowTooltipSoft = (show: boolean) => {
+    setShowTooltip(show);
+  };
+
+  const onShowTooltipHard = (show: boolean) => {
+    setShowTooltip(show);
+  };
+
+  useEffect(() => {
+    if (hasTooltip && tooltipValue) {
+      const close = () => {
+        setShowTooltip(false);
+      };
+
+      const handlePointerDownEvent = (event: PointerEvent) => {
+        if (event.target && containerRef && (containerRef.current as any)?.contains(event.target as HTMLElement)) {
+          return;
+        }
+        close();
+      };
+
+      const handleKeyDownEvent = (event: KeyboardEvent) => {
+        if (event.key === 'Escape') {
+          close();
+        }
+      };
+
+      window.addEventListener('pointerdown', handlePointerDownEvent);
+      window.addEventListener('keydown', handleKeyDownEvent);
+
+      return () => {
+        window.removeEventListener('pointerdown', handlePointerDownEvent);
+        window.removeEventListener('keydown', handleKeyDownEvent);
+      };
+    }
+  }, [containerRef, hasTooltip, tooltipValue]);
+
   return (
-    <div className={clsx(styles['trigger-wrapper'], !highContrastHeader && styles['remove-high-contrast-header'])}>
-      <button
-        aria-expanded={ariaExpanded}
-        aria-controls={ariaControls}
-        aria-haspopup={true}
-        aria-label={ariaLabel}
-        className={clsx(
-          styles.trigger,
-          {
-            [styles.selected]: selected,
-            [styles.badge]: badge,
-          },
-          className
-        )}
-        onClick={onClick}
-        ref={ref as React.Ref<HTMLButtonElement>}
-        type="button"
-        data-testid={testId}
-      >
-        <span className={clsx(badge && styles['trigger-badge-wrapper'])}>
-          <Icon name={iconName} svg={iconSvg} />
-        </span>
-      </button>
-      {badge && <div className={styles.dot} />}
+    <div
+      ref={containerRef}
+      {...(hasTooltip
+        ? {
+            onPointerEnter: () => onShowTooltipSoft(true),
+            onPointerLeave: () => onShowTooltipSoft(false),
+            onFocus: ({ target, currentTarget }) => {
```

Comment:
I'm looking into creating a custom focus handler that will check for details in the event, and not show the tooltip is a detail stating it was triggered from a drawer close via the close button on the drawer